### ISSUE TITLE
Some Task Groups don't Expand when Clicked

### ISSUE THREAD

**Issue by MatrixManAtYrService** [apache/airflow#18662]
### Apache Airflow version

2.2.0b2 (beta snapshot)

### Operating System

debian (docker)

### Versions of Apache Airflow Providers

n/a

### Deployment

Astronomer

### Deployment details

`astro dev start`

Dockerfile:
```
FROM quay.io/astronomer/ap-airflow-dev:2.2.0-buster-44417
```

dag:
https://gist.github.com/MatrixManAtYrService/a3ee53dab27edf8249d0ff0dfd11c76e
![output](https://user-images.githubusercontent.com/5834582/135555462-e129eaa3-a24c-4c5f-8e29-534fca10b987.png)

The image above was generated with the command:
```
airflow dags show structure_hash | grep digraph -A999 | dot -Tpng  > output.png
```

### What happened

I have other DAGs with task groups, and they expand/collapse just fine, but this one refuses to expand when I click its blue box.

<img width="978" alt="Screen Shot 2021-09-30 at 8 22 35 PM" src="https://user-images.githubusercontent.com/5834582/135555917-b2baa60e-f3bf-4dc1-bffb-6ea6b18d59b6.png">

Instead my browser's js console shows an error:
```
Uncaught SyntaxError: Invalid or unexpected token
dagre-d3.min.js:1642 Uncaught TypeError: Cannot convert undefined or null to object
    at Graph._removeFromParentsChildList (dagre-d3.min.js:1642)
    at Graph.setParent (dagre-d3.min.js:1642)
    at graph.d7b2753a50a36b9c46ed.js:1485
    at Array.forEach (<anonymous>)
    at expandGroup (graph.d7b2753a50a36b9c46ed.js:1470)
    at SVGGElement.<anonymous> (graph.d7b2753a50a36b9c46ed.js:1044)
    at SVGGElement.__onclick (d3.min.js:1)
```

### What you expected to happen

The blue boxen expand when clicked, ultimately exposing the structure show in the **Deployment Details** section.

### How to reproduce

Load the dag mentioned above, navigate to its graph view, try to expand the task groups

### Anything else

Sorry the DAG is so weird.  I spent a little time trying to cobble together components that reproduced this more minimally, but I didn't stumble across whatever secret sauce makes this one special.

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by bbovenzi** [comment]
Thanks for providing a DAG. That will make it easier to figure out this odd issue.

**Comment by bbovenzi** [comment]
Interestingly, they render just fine if they start expanded and if you close it. The only problem is opening them. I'll keep investigating.

<img width="377" alt="Screen Shot 2021-10-04 at 12 34 10 PM" src="https://user-images.githubusercontent.com/4600967/135836721-dbdacf78-83e1-4b0b-adc4-6983460b1641.png">

**Comment by MatrixManAtYrService** [comment]
I spent a while trying to reduce DAG complexity towards a more minimal case, and I was able to get them quite minimal indeed while still reproducing the issue.  But then later when I tried starting from nothing and building *up* to the minimal case I couldn't reproduce it--which is why I ended up posting the whole thing.

My feeling was that something was being remembered between iterations that I didn't expect to be.  I didn't take careful notes during this exploration though, so it remains just a feeling.
